File: stress_test.py
import time
import concurrent.futures
import requests

URL = "https://sanjeevkumartiwari.com"
CONCURRENCY = 10
TOTAL_REQUESTS = 100

def fetch_url(url):
    try:
        start_time = time.time()
        response = requests.get(url, timeout=10)
        end_time = time.time()
        return response.status_code, end_time - start_time
    except Exception as e:
        return "ERROR", str(e)

def run_stress_test():
    print(f"Starting Stress Test on {URL}...")
    print(f"Concurrency: {CONCURRENCY}, Total Requests: {TOTAL_REQUESTS}")
    
    results = []
    with concurrent.futures.ThreadPoolExecutor(max_workers=CONCURRENCY) as executor:
        future_to_url = {executor.submit(fetch_url, URL): i for i in range(TOTAL_REQUESTS)}
        for future in concurrent.futures.as_completed(future_to_url):
            results.append(future.result())

    success_codes = [r[0] for r in results if isinstance(r[0], int) and r[0] == 200]
    total_time = [r[1] for r in results if isinstance(r[1], float)]
    
    avg_time = sum(total_time) / len(total_time) if total_time else 0
    success_rate = (len(success_codes) / TOTAL_REQUESTS) * 100
    
    print("\n--- RESULTS ---")
    print(f"Total Requests: {TOTAL_REQUESTS}")
    print(f"Succesful (200 OK): {len(success_codes)}")
    print(f"Success Rate: {success_rate}%")
    print(f"Average Response Time: {avg_time:.4f}s")
    print(f"Fastest Request: {min(total_time) if total_time else 0:.4f}s")
    print(f"Slowest Request: {max(total_time) if total_time else 0:.4f}s")

if __name__ == "__main__":
    run_stress_test()

File: stress_test.js
import https from 'https';

const URL = "https://sanjeevkumartiwari.com";
const CONCURRENCY = 50;
const TOTAL_REQUESTS = 1000;

async function fetchUrl(url) {
    return new Promise((resolve) => {
        const start = Date.now();
        const req = https.get(url, (res) => {
            res.on('data', () => { }); // Consume stream
            res.on('end', () => {
                resolve({ status: res.statusCode, time: (Date.now() - start) / 1000 });
            });
        });

        req.on('error', (e) => {
            resolve({ status: "ERROR", error: e.message });
        });

        req.setTimeout(15000, () => {
            req.destroy();
            resolve({ status: "TIMEOUT" });
        });
    });
}

async function runStressTest() {
    process.stdout.write(`--- 1000 USERS SIMULATION --- \n`);
    process.stdout.write(`URL: ${URL}\n`);
    process.stdout.write(`Batch Size (Concurrency): ${CONCURRENCY}\n`);
    process.stdout.write(`Total simulated users: ${TOTAL_REQUESTS}\n\n`);

    const results = [];

    for (let i = 0; i < TOTAL_REQUESTS; i += CONCURRENCY) {
        process.stdout.write(`Sending batch ${i / CONCURRENCY + 1} of ${TOTAL_REQUESTS / CONCURRENCY}...\n`);
        const batch = [];
        for (let j = 0; j < CONCURRENCY && (i + j) < TOTAL_REQUESTS; j++) {
            batch.push(fetchUrl(URL));
        }
        const bResults = await Promise.all(batch);
        results.push(...bResults);
    }

    const successCodes = results.filter(r => r.status === 200);
    const errors = results.filter(r => r.status === "ERROR" || r.status === "TIMEOUT");
    const times = results.map(r => r.time).filter(t => typeof t === 'number');

    const avgTime = times.reduce((a, b) => a + b, 0) / times.length;
    const successRate = (successCodes.length / TOTAL_REQUESTS) * 100;

    process.stdout.write("\n--- FINAL SITE AUDIT RESULTS ---\n");
    process.stdout.write(`Total Load Simulations: ${TOTAL_REQUESTS}\n`);
    process.stdout.write(`Succesful Responses (200 OK): ${successCodes.length}\n`);
    process.stdout.write(`Failures/Timeouts: ${errors.length}\n`);
    process.stdout.write(`Success Rate: ${successRate.toFixed(2)}%\n`);
    process.stdout.write(`Average UX Response: ${avgTime.toFixed(4)}s\n`);
    process.stdout.write(`P95 Latency (Est): ${times.sort((a, b) => a - b)[Math.floor(times.length * 0.95)]?.toFixed(4)}s\n`);
}

runStressTest();
